src/main/kotlin/org/team5419/fault/BerkeliumIterativeRobotBase.kt

```diff
@@ -0,0 +1,139 @@
+package org.team5419.fault
+
+import edu.wpi.first.hal.HAL
+import edu.wpi.first.wpilibj.RobotBase
+import edu.wpi.first.wpilibj.Timer
+import org.team5419.fault.math.units.SIUnit
+import org.team5419.fault.math.units.Second
+import org.team5419.fault.math.units.seconds
+
+@Suppress("TooManyFunctions")
+abstract class BerkeliumIterativeRobotBase(
+    protected var mPeriod: SIUnit<Second>,
+    protected var mOverrunPrintPeriod: SIUnit<Second>
+) : RobotBase() {
+
+    enum class Mode {
+        None,
+        Disabled,
+        Autonomous,
+        Teleop,
+        Test
+    }
+
+    private var mLastMode = Mode.None
+
+    private var mLoopStartTime = 0.0.seconds
+    private var mLoopEndTime = 0.0.seconds
+
+    private var mLastOverrunDump = 0.0.seconds
+    private var mOverrunCount = 0
+
+    abstract override fun startCompetition()
+
+    // init methods
+    open fun robotInit() = println("Default robot init method... Override me")
+    open fun disabledInit() = println("Default disabled init method... Override me")
+    open fun teleopInit() = println("Default teleop init method... Override me")
+    open fun autonomousInit() = println("Default auto init method... Override me")
+    open fun testInit() = println("Default test init method... Override me")
+
+    // periodic loop message guards
```

Comment:
change this comment so that instead of saying what the variables are, it says how they are used / why they are needed